### PR TITLE
Introduce Poetry and bump up e2e

### PR DESCRIPTION
**Pull request by Superskyyy**
A step in preparing the 1.0.0 release.

* Replace venv/pip with Poetry
* Bump up e2e dependencies (OAP, swctl)
* Support slim python docker image
	Slim docker images are tested in CI and produced by the `make build-image` process.

The following changes are mostly side effects encountered during the migration.

* Remove Python 3.6 from the testing matrix (removal of support will come in the next PR)
	Poetry and planned os.register_at_fork API usage in agent requires 3.7+, and 3.6 has reached end of support.

* Update documentation to reflect poetry migration
	New docs on how to setup development env  `make env`, `make test` etc.

* Update PR template
	Reflect ^ above changes.

* Bump up swctl & OAP version in e2e
	Bump up to 9.2.0 

* fix log list e2e, trace list e2e, service instance (.total, errorreason, layer concept)
	Reflect changes in expected results due to the bump up.

* fix kafka ignore (parenthesis missing)
	Kafka plugin self data ignore 

* fix meter e2e (naming)
	Fix meter e2e thanks to the help from @jiang1997 (https://github.com/apache/skywalking/pull/9912)

### CODE REVIEW

**Comment by Superskyyy** [comment]
@jiang1997 if you have time, please help to try the changes out (the poetry part), `make env` command etc. And see if they work in your environment.

**Comment by jiang1997** [comment]
> @jiang1997 if you have time, please help to try the changes out (the poetry part), `make env` command etc. And see if they work in your environment.

Got it.

**Comment by kezhenxu94** [comment]
I cannot run this and failed to find a workable solution in Google, I did found some similar questions/issues but none works for me. Haven't dig deep to see why and how to fix this yet.

```
skywalking-python pr/Superskyyy/249 % make
curl -sSL https://install.python-poetry.org | python3 -
Retrieving Poetry metadata

# Welcome to Poetry!

This will download and install the latest version of Poetry,
a dependency and package manager for Python.

It will add the `poetry` command to Poetry's bin directory, located at:

/Users/kezhenxu94/.local/bin

You can uninstall at any time by executing this script with the --uninstall option,
and these changes will be reverted.

Installing Poetry (1.2.2): Creating environment
Traceback (most recent call last):
  File "<stdin>", line 940, in <module>
  File "<stdin>", line 919, in main
  File "<stdin>", line 550, in run
  File "<stdin>", line 571, in install
  File "/Applications/Xcode.app/Contents/Developer/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/contextlib.py", line 117, in __enter__
    return next(self.gen)
  File "<stdin>", line 643, in make_env
  File "<stdin>", line 629, in make_env
  File "<stdin>", line 309, in make
  File "/Applications/Xcode.app/Contents/Developer/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/venv/__init__.py", line 66, in __init__
    self.symlinks = should_use_symlinks(symlinks)
  File "/Applications/Xcode.app/Contents/Developer/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/venv/__init__.py", line 31, in should_use_symlinks
    raise Exception("This build of python cannot create venvs without using symlinks")
Exception: This build of python cannot create venvs without using symlinks
make: [poetry] Error 1 (ignored)
poetry self update
make: poetry: No such file or directory
make: *** [poetry] Error 1
skywalking-python pr/Superskyyy/249 % 
```

**Comment by Superskyyy** [comment]
> I cannot run this and failed to find a workable solution in Google, I did found some similar questions/issues but none works for me. Haven't dig deep to see why and how to fix this yet.
> 
Seems like a macOS's built-in python issue here https://github.com/python-poetry/install.python-poetry.org/issues/24

There isn't a final fix yet, but installing poetry from homebrew looks like an easy workaround for now. 
Can we do something like this?

```makefile
ifeq ($(OS),Windows_NT)
    OS := Windows
else ifeq ($(shell uname -s),Darwin)
    OS := Darwin
else
    OS := $(shell sh -c 'uname 2>/dev/null || echo Unknown')
endif

.PHONY: poetry

poetry:
ifeq ($(OS),Windows)
	-powershell (Invoke-WebRequest -Uri https://install.python-poetry.org -UseBasicParsing).Content | py -
	poetry self update
else ifeq ($(OS),Darwin)
	brew install poetry
	poetry self update
else
	-curl -sSL https://install.python-poetry.org | python3 -
	poetry self update
endif
```

**Comment by kezhenxu94** [comment]
> > I cannot run this and failed to find a workable solution in Google, I did found some similar questions/issues but none works for me. Haven't dig deep to see why and how to fix this yet.
> 
> Seems like a macOS's built-in python (not for other python versions) issue here [python-poetry/install.python-poetry.org#24](https://github.com/python-poetry/install.python-poetry.org/issues/24)
> 
> There isn't a final fix yet, but installing poetry from homebrew looks like an easy workaround for now. (Can you see if it works?) If so we could do something like this until they fix it?
> 
> ```makefile
> ifeq ($(OS),Windows_NT)
>     OS := Windows
> else
>     OS := $(shell sh -c 'uname 2>/dev/null || echo Unknown')
> endif
> 
> .PHONY: poetry
> 
> poetry:
> ifeq ($(OS),Windows)
> 	-powershell (Invoke-WebRequest -Uri https://install.python-poetry.org -UseBasicParsing).Content | py -
> 	poetry self update
> else ifeq ($(OS),Darwin)
> 	brew install poetry
> 	poetry self update
> else
> 	-curl -sSL https://install.python-poetry.org | python3 -
> 	poetry self update
> endif
> ```

I'm not sure whether that works for everyone, but at least it doesn't work for me, I don't have `homebrew` installed.

**Comment by Superskyyy** [comment]
> > > I cannot run this and failed to find a workable solution in Google, I did found some similar questions/issues but none works for me. Haven't dig deep to see why and how to fix this yet.
> > 

> > ```
> 
> I'm not sure whether that works for everyone, but at least it doesn't work for me, I don't have `homebrew` installed.

My bad I thought homebrew come in macOS as preinstalled. Then we have to use this then, which is among the official ways of installing poetry.

```
python3 -m pip install --user pipx
python3 -m pipx ensurepath
pipx install poetry
pipx upgrade poetry
```

**Comment by kezhenxu94** [comment]
> > > > I cannot run this and failed to find a workable solution in Google, I did found some similar questions/issues but none works for me. Haven't dig deep to see why and how to fix this yet.
> 
> > > 
> 
> 
> 
> > > ```
> 
> > 
> 
> > I'm not sure whether that works for everyone, but at least it doesn't work for me, I don't have `homebrew` installed.
> 
> 
> 
> My bad I thought homebrew come in macOS as preinstalled. Then we have to use this then, which is among the official ways of installing poetry.
> 
> 
> 
> ```
> 
> python3 -m pip install --user pipx
> 
> python3 -m pipx ensurepath
> 
> pipx install poetry
> 
> pipx upgrade poetry
> 
> ```
> 
> 

This looks good.

File: README.md [hunk 1]
```diff
@@ -2,15 +2,17 @@
...
+**SkyWalking-Python**: The Python Agent for Apache SkyWalking provides the native tracing/metrics/logging abilities for Python projects.
+
+**[SkyWalking](https://github.com/apache/skywalking)**: Application performance monitor tool for distributed systems, especially designed for microservices, cloud native and container-based (Kubernetes) architectures.
 
-**SkyWalking**: an APM(application performance monitor) system, especially designed for
-microservices, cloud native and container-based (Docker, Kubernetes, Mesos) architectures.
 
 [![GitHub stars](https://img.shields.io/github/stars/apache/skywalking-python.svg?style=for-the-badge&label=Stars&logo=github)](https://github.com/apache/skywalking-python)
-[![Twitter Follow](https://img.shields.io/twitter/follow/asfskywalking.svg?style=for-the-badge&label=Follow&logo=twitter)](https://twitter.com/AsfSkyWalking)
+[![Twitter Follow](https://imQDg.shields.io/twitter/follow/asfskywalking.svg?style=for-the-badge&label=Follow&logo=twitter)](https://twitter.com/AsfSkyWalking)
```
Comment (kezhenxu94) [inline README.md:11]: Is this changed by mistake?

File: tests/plugin/Dockerfile.plugin [hunk 2]
```diff
@@ -23,6 +23,6 @@ WORKDIR /agent
 
 ADD $ROOT /agent
 
-RUN cd /agent && make setup-test
+RUN cd /agent && make install
 
-ENV PATH="/agent/venv/bin:$PATH"
+# ENV PATH="/agent/venv/bin:$PATH"
```
Comment (kezhenxu94) [inline tests/plugin/Dockerfile.plugin:28]:
```suggestion
```

File: README.md (hunk 1 above)
Comment (Superskyyy) [reply README.md:11]: Oops, IDE turned on me.

File: .github/PULL_REQUEST_TEMPLATE [hunk 3]
```diff
@@ -1,9 +1,13 @@
-<!-- Uncomment the following checklist WHEN AND ONLY WHEN you're adding a new plugin -->
+<!-- Fill the following checklists when a feature/plugin is added or modified-->
```
[the remark this answers is not in the document]
Comment (Superskyyy) [reply .github/PULL_REQUEST_TEMPLATE]: Thank you, now I suppose using the main repo's PR template will be a better choice, I added an additional plugin section there.

[the remark this answers is not in the document]
Comment (Superskyyy) [reply .github/PULL_REQUEST_TEMPLATE]: Changes pushed.

**Review by kezhenxu94 — APPROVED** [review]
Thanks!

**Pull request merged**